.github/workflows/static-analysis.yml

```diff
@@ -0,0 +1,103 @@
+name: Static Analysis Check
+
+on:
+  pull_request:
+  push:
+#    branches:
+#      - main
+#      - master
+
+env:
+  BUILD_TYPE: Debug
+  # Conan cache environment variables
+  CONAN_SYSREQUIRES_MODE: enabled
+  CONAN_USER_HOME: "${{ github.workspace }}/conan-cache"
+  CONAN_USER_HOME_SHORT: "${{ github.workspace }}/conan-cache/short"
+
+  # Customize clang
+  USE_NEWER_CLANG: false    # Set this to install clang newer than the default version
+
+  # Constants for include-what-you-use
+  IWYU_BUILD: /home/iwyu/build
+  IWYU_SRC: /home/iwyu/include-what-you-use
+  IWYU_GIT_URL: https://github.com/include-what-you-use/include-what-you-use.git
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build:
+    runs-on: 'ubuntu-latest'
+
+    steps:
+      - name: Install python dependencies
+        run: |
+          python --version | grep -iE "python\s*3"  # fail if python 2 is default
+          pip    --version | grep -iE "python\s*3"
+          pip install cmake conan
+
+      - name: Install cppcheck
+        run: |
+          sudo apt-get update -qq
+          sudo apt-get install cppcheck
+
+      - name: Install default clang (clang-10 on Ubuntu 20.04)
+        if: env.USE_NEWER_CLANG == false
+        run: |
+          sudo apt get install clang llvm-dev libclang-dev clang-tidy
+
+      # TODO: use GH cache for clang and IWYU
+      - name: Install newer clang
+        if: env.USE_NEWER_CLANG == true
+        env:
+          LLVM_VER: 12
+        run: |
+          su && echo -e '\U0001F9B8' # 🦸
+          wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add - 2>/dev/null
+          add-apt-repository -y "deb https://apt.llvm.org/bionic/ llvm-toolchain-bionic-${LLVM_VER} main"
+          apt-get update -qq
+          apt-get install -y --no-install-recommends \
+            clang-${LLVM_VER} lldb-${LLVM_VER} lld-${LLVM_VER} clangd-${LLVM_VER} \
+            llvm-${LLVM_VER}-dev libclang-${LLVM_VER}-dev clang-tidy-${LLVM_VER}
+          update-alternatives --install /usr/bin/clang-tidy clang-tidy $(which clang-tidy-${LLVM_VER}) 10
+          update-alternatives --install /usr/bin/clang   clang   $(which clang-${LLVM_VER}) 10
+          update-alternatives --install /usr/bin/clang++ clang++ $(which clang++-${LLVM_VER}) 10
```

Comment:
Please use https://github.com/kylemayes/install-llvm-action instead. You can use it to easily install LLVM.

To setup libclang, you only need this:
https://github.com/kylemayes/install-llvm-action#linking-to-installed-libraries-linux

Reply:
The real problem here is IWYU. That's the only reason to install any of this stuff. There is an [action](https://github.com/emilgedda/include-what-you-use-action) you can use to install it, but it locks you into using clang-9. It won't work if you need to use newer C++ features.

I think we want to be able to build and install IWYU with an arbitrary version of clang and cache the result of that process. IWYU takes much longer to build than it takes to install any of the clang packages.

If you're interested in taking over that job, I welcome the effort. I'm not going to have time to do it myself for a while.